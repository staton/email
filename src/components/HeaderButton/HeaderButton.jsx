import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

const propTypes = {
    content: PropTypes.any.isRequired,
    id: PropTypes.string.isRequired,
    onClick: PropTypes.func
};

const defaultProps = {
    onClick: () => {}
};

export class HeaderButton extends Component {
	render() {
		return (
            <button 
                className="HeaderButton"
                id={this.props.id}
                onClick={this.props.onClick}>
				{this.props.content}
			</button>
		);
	}
}

function mapStateToProps(store, ownProps) {
    return {
        content: ownProps.content,
        onClick: ownProps.onClick
    };
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators({
    },
    dispatch);
}

HeaderButton.propTypes = propTypes;
HeaderButton.defaultProps = defaultProps;

export default connect(mapStateToProps, mapDispatchToProps)(HeaderButton);