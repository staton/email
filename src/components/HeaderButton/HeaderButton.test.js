import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import { HeaderButton } from './HeaderButton';

describe('<HeaderButton />', () => {

    const GET_PROPS = () => {
        content: "click me"
    };

    Enzyme.configure({ adapter: new Adapter() });

    it('renders correctly', () => {
        const wrapper = shallow(<HeaderButton {...GET_PROPS()}/>);
        expect(wrapper).toMatchSnapshot();
    });
    
});