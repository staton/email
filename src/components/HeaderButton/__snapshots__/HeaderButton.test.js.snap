// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderButton /> renders correctly 1`] = `
<button
  className="HeaderButton"
  id="btn-id"
  onClick={[Function]}
>
  X
</button>
`;
